src/utils/string/StringUtils.js

```diff
@@ -508,6 +511,11 @@ const getCaptionWithTopicStyle = (
       isShortText={isShort}
     />
   ));
+
+  text = reactStringReplace(text, urlRegex, (match) => {
+    return <LinkableText text={match} withTopicDetection={true} />;
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `LinkableText` component to replace URLs in the text with clickable links is a good addition. However, ensure that the `LinkableText` component properly handles invalid or broken URLs to prevent any potential crashes or unexpected behavior.
```diff
  text = reactStringReplace(text, urlRegex, (match) => {
-     return <LinkableText text={match} withTopicDetection={true} />;
+     return <LinkableText text={match} withTopicDetection={true} onError={handleError} />;
  });
```
In the above diff, `handleError` is a hypothetical function that you would need to implement to handle errors when a link is invalid or broken.


<!-- This is an auto-generated comment by OSS CodeRabbit -->